Sources/WebSocket/NWError+WebSocket.swift

```diff
@@ -0,0 +1,53 @@
+import Foundation
+import Network
+import WebSocketProtocol
+
+extension NWError {
+    var isNotConnected: Bool {
+        switch self {
+        case .posix(.ENOTCONN): return true
+        default: return false
+        }
+    }
+
+    var isCancelled: Bool {
+        switch self {
+        case .posix(.ECANCELED): return true
+        default: return false
+        }
+    }
+
+    var shouldCloseConnectionWhileConnectingOrOpen: Bool {
+        switch self {
+        case .posix(.ECANCELED):
+            return false
+        case .posix(.ENOTCONN):
+            return false
+        default:
+            print("Unhandled error in '\(#function)': \(debugDescription)")
+            return true
+        }
+    }
+}
+
+extension NWError {
+    var webSocketError: WebSocketError {
+        switch self {
+        case .posix(.ECANCELED):
+            return .closed(.normalClosure, Data(localizedDescription.utf8))
+        default:
+            print("Unhandled error in '\(#function)': \(debugDescription)")
+            return .closed(.normalClosure, Data(localizedDescription.utf8))
+        }
+    }
+
+    var closeCode: WebSocketCloseCode {
+        switch self {
+        case .posix(.ECANCELED):
+            return .normalClosure
+        default:
+            print("Unhandled error in '\(#function)': \(debugDescription)")
+            return .normalClosure
+        }
+    }
+}
```

Comment:
How did you arrive at this extension?

Reply:
I don't know if we need this extension, but our public protocol abstracts away errors behind a non-framework-specific `WebSocketError` type. It makes sense to map as many `NWError` types to that `WebSocketError` type as possible to present a simple, unified API to consumers.

Stepping back a bit, the error handling for `NWConnection` is a bit overwhelming. `NWError` is comprehensive, which is great, but it includes tons and tons of potential errors, only some of which _(I think and hope)_ are applicable to WebSockets. I want to figure out which `NWError` cases I need to worry about and deal with them here, in one location. This extension is by no means finished. I'm logging "Unhandled errors" because I want to figure out which sets of errors actually appear during the use of `NWConnection` and then deal with them specifically.